src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_shp/LocalServerDynamicWorkspaceShapefileSample.java

```diff
@@ -0,0 +1,172 @@
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Dynamic workspaces: shapefile");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create Choose Shapefile button
+      Button addButton = new Button("Choose Shapefile");
+      addButton.setMaxSize(150, 25);
+      addButton.setDisable(true);
+
+      // start local server if found
+      if (LocalServer.INSTANCE.checkInstallValid()) {
+        server = LocalServer.INSTANCE;
+        server.addStatusChangedListener(status -> {
+          if (server.getStatus() == LocalServerStatus.STARTED) {
+            addButton.setDisable(false);
+          }
+        });
+        // start local server
+        server.startAsync();
+      } else {
+        Platform.runLater(() -> {
+          Alert dialog = new Alert(AlertType.INFORMATION);
+          dialog.setHeaderText("Local Server Load Error");
+          dialog.setContentText("Local Server install path couldn't be located.");
+          dialog.showAndWait();
+          Platform.exit();
+        });
+      }
+
+      // choose the file, then start the local map service
+      addButton.setOnAction(e -> {
+        // Browse to the shapefile file
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Open Resource File");
+        fileChooser.getExtensionFilters().addAll(new ExtensionFilter("Shapefiles", "*.shp"));
+        fileChooser.setInitialDirectory(new File("./samples-data/shapefiles/"));
+        File selectedFile = fileChooser.showOpenDialog(stage);
+
+        if (selectedFile != null) {
+          String fileName = selectedFile.getName();
+          String path = selectedFile.getParent();
+          startLocalMapService(fileName, path);
+        }
+      });
+
+      // create a map and add it to a map view
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add the map view and control panel to stack pane
+      stackPane.getChildren().addAll(mapView, addButton);
+      StackPane.setAlignment(addButton, Pos.TOP_LEFT);
+      StackPane.setMargin(addButton, new Insets(10, 0, 0, 10));
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Start a LocalMapService and attaches a dynamic workspace shapefile.
+   * 
+   * @param fileName
+   * @param path
+   */
+  private void startLocalMapService(String fileName, String path) {
+
+    // start a service from the blank MPK
+    String mapServiceURL = "./samples-data/local_server/mpk_blank.mpk";
+    LocalMapService localMapService = new LocalMapService(mapServiceURL);
+
+    // Can't add a dynamic workspace to a running service, so do that first
```

Comment:
less colloquial language. should be more direct to follow the pattern set in other samples. Just `//create a shapefile workspace`.